app/models/order_detail.rb

```diff
@@ -3,4 +3,22 @@
 class OrderDetail < ApplicationRecord
   belongs_to :order
   belongs_to :book
+
+  scope :top_total, lambda { |year|
+    joins(:order).where("YEAR(orders.created_at) = ? AND orders.status = 1", year)
+                 .group("(order_details.book_id)").order("total desc")
+                 .pluck("SUM(order_details.quantity) as total").uniq[4]
```

Comment:
dùng uniq có thấy warning này không
DEPRECATION WARNING: uniq is deprecated and will be removed from Rails 5.1 (use distinct instead) (called from […])

Reply:
dạ có anh, để em update lại =))